src/main/java/com/atypon/wayf/facade/impl/PublisherFacadeImpl.java

```diff
@@ -115,7 +116,22 @@ public PublisherFacadeImpl() {
 
     @Override
     public Observable<Publisher> filter(PublisherQuery filter) {
-        return publisherDao.filter(filter);
+
+        boolean isAdminView = PublisherQuery.ADMIN_VIEW.equals(filter.getView());
+
+        if (isAdminView) {
+            AuthenticatedEntity.authenticatedAsAdmin(RequestContextAccessor.get().getAuthenticated());
+        }
+
+        return publisherDao.filter(filter).flatMap(publisher -> {
+            if (!isAdminView) {
+                publisher.setWidgetLocation(null);
+            } else {
+                publisher.setToken((AuthorizationToken) authenticationFacade.getCredentialsForAuthenticatable(publisher)
+                        .takeWhile(token -> token instanceof AuthorizationToken).blockingFirst());
```

Comment:
just a note, you might want to avoid the blocking calls because the interrupt the flow of events. Here, you could used .first() and then FacadePolicies.singleOrException to get that as a single and then map that to something. You wouldn't have to have Observable.just at the end of this. 

The blockingFirst and blockingGet calls can throw exceptions and they might not be handled uniformly. This is just a broad piece of advice that I had issues getting clean stack traces out of nested blockingGets because they exception breaks the chain. We can discuss this on a call, it's a little weird.